src/sentry/dynamic_sampling/tasks.py

```diff
@@ -0,0 +1,80 @@
+import logging
+from typing import Mapping, Sequence
+
+from sentry import features
+from sentry.dynamic_sampling.models.adjustment_models import AdjustedModel
+from sentry.dynamic_sampling.models.adjustment_models import Project as DSProject
+from sentry.dynamic_sampling.prioritise_projects import fetch_projects_with_total_volumes
+from sentry.dynamic_sampling.rules.helpers.prioritise_project import _generate_cache_key
+from sentry.dynamic_sampling.rules.helpers.utils import get_redis_client_for_ds
+from sentry.models import Organization
+from sentry.tasks.base import instrumented_task
+from sentry.tasks.relay import schedule_invalidate_project_config
+from sentry.utils import metrics
+
+CHUNK_SIZE = 1000
+MAX_SECONDS = 60
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.tasks.prioritise_projects",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+)  # type: ignore
+def prioritise_projects() -> None:
+    metrics.incr("sentry.tasks.dynamic_sampling.prioritise_projects.start", sample_rate=1.0)
+    with metrics.timer("sentry.tasks.dynamic_sampling.prioritise_projects", sample_rate=1.0):
+        for org_id, project_id_with_count_per_root in fetch_projects_with_total_volumes().items():
```

Comment:
`project_id_with_count_per_root` sounds a bit confusing to me, I would expect a plural name for a collection. What about `projects_with_tx_count`?

Reply:
yeah it's a good rename done.